persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/slick/SlickReadSideImpl.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.scaladsl.persistence.slick
+
+import akka.{ Done, NotUsed }
+import akka.persistence.query.Offset
+import akka.stream.scaladsl.Flow
+import com.lightbend.lagom.internal.persistence.jdbc.{ SlickOffsetDao, SlickOffsetStore, SlickProvider }
+import com.lightbend.lagom.scaladsl.persistence.ReadSideProcessor.ReadSideHandler
+import com.lightbend.lagom.scaladsl.persistence.slick.SlickReadSide
+import com.lightbend.lagom.scaladsl.persistence.{ AggregateEvent, AggregateEventTag, EventStreamElement }
+import org.slf4j.LoggerFactory
+import slick.dbio.{ DBIOAction, NoStream }
+import slick.jdbc.JdbcProfile
+import slick.jdbc.JdbcBackend.Database
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.reflect.ClassTag
+
+class SlickReadSideImpl(slick: SlickProvider, offsetStore: SlickOffsetStore)(implicit val executionContext: ExecutionContext)
+  extends SlickReadSide {
+
+  override val db: Database = slick.db
+  override val profile: JdbcProfile = slick.profile
+
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  override def builder[Event <: AggregateEvent[Event]](readSideId: String): ReadSideHandlerBuilder[Event] = new ReadSideHandlerBuilder[Event] {
+    var globalPrepare: DBIOAction[Any, _, _] = DBIOAction.successful(())
+    var prepare: (AggregateEventTag[Event]) => DBIOAction[Any, NoStream, Nothing] = (_) => DBIOAction.successful(())
+    var eventHandlers = Map.empty[Class[_ <: Event], (EventStreamElement[_ <: Event]) => DBIOAction[Any, NoStream, Nothing]]
+
+    override def setGlobalPrepare(callback: DBIOAction[Any, _, _]): ReadSideHandlerBuilder[Event] = {
```

Comment:
`DBIOAction` is fundamentally a wrapper for `Connection => T`. That is to say, it is a callback.  In fact, you can create one like this:

```scala
setGlobalPrepare(SimpleDBIO { connection =>
  // do anything, regardless of whether you use the connection or not
})
```

I think if you're a Slick user, if you don't already understand this, you will need to pretty quickly anyway, it's pretty fundamental to understanding Slick, so I don't think Lagom is adding any burden on the user of things they need to understand here.  So I think it makes sense to use `DBIOAction` in this way.